### PR TITLE
DNM: duplicate upgrade warning code

### PR DESCRIPTION
**Pull request by shinobu-x**
Clean up: src/mon/OSDMonitor.cc: duplicate upgrade warning code

Tracker: http://tracker.ceph.com/issues/17749
Signed-off-by: Shinobu Kinjo <shinobu@redhat.com>

### CODE REVIEW

**Comment by shinobu-x** [comment]
If anyone help me reduce duplicated strings in the message (e.g., ``all OSDs are running``), it would be appreciated. And if it could be able to become something which has broad utility, it would be super great help :)

File: src/mon/OSDMonitor.cc [hunk 1]
```diff
@@ -3115,30 +3115,28 @@ void OSDMonitor::get_health(list<pair<health_status_t,string> >& summary,
...
+      !osdmap.test_flag(CEPH_OSDMAP_REQUIRE_KRAKEN)) ?
+      f = "kraken" : f = "";
+    (HAVE_FEATURE(osdmap.get_up_osd_features(), SERVER_JEWEL) &&
+      !osdmap.test_flag(CEPH_OSDMAP_REQUIRE_JEWEL)) ?
+      f = "jewel" : f = "";
+
+    if (!f.empty()) {
+
+      m = "all OSDs are running " + f + " or later but the"
+        " '" + f + "' osdmap flag is not set";
```
Comment (joscollin) [inline src/mon/OSDMonitor.cc]: The resulting string in m is not the same as the previous string in msg.

File: src/mon/OSDMonitor.cc [hunk 2]
```diff
@@ -3115,30 +3115,28 @@ void OSDMonitor::get_health(list<pair<health_status_t,string> >& summary,
...
-      string msg = "all OSDs are running jewel or later but the"
-	" 'require_jewel_osds' osdmap flag is not set";
-      summary.push_back(make_pair(HEALTH_WARN, msg));
-      if (detail) {
-	detail->push_back(make_pair(HEALTH_WARN, msg));
-      }
+
+    std::string m, f;
+
+    (HAVE_FEATURE(osdmap.get_up_osd_features(), SERVER_LUMINOUS) &&
```
Comment (joscollin) [inline src/mon/OSDMonitor.cc]: The code is better now. It avoids repeated code. But what is the point in removing the if-elseif ?

File: src/mon/OSDMonitor.cc [hunk 3]
```diff
@@ -3115,30 +3115,28 @@ void OSDMonitor::get_health(list<pair<health_status_t,string> >& summary,
...
+      !osdmap.test_flag(CEPH_OSDMAP_REQUIRE_LUMINOUS)) ?
+      f = "luminous" : f = "";
```
Comment (liewegas) [inline src/mon/OSDMonitor.cc]: each of these should just be an if (or else if) and set f = whatever; no need for the awkward ? operator

File: src/mon/OSDMonitor.cc (hunk 1 above)
Comment (liewegas) [inline src/mon/OSDMonitor.cc]: and here it's 
... " 'require_" + f + "_osds'" ...

**Comment by shinobu-x** [comment]
@liewegas @joscollin Thanks for your comments. Much appreciated.
I've not been convinced by what I modified which is stupid modification... I need to brush up my brain more.

**Comment by liewegas** [comment]
DNM = do not merge

**Comment by joscollin** [comment]
@liewegas  OK, I'm sorry if it is meant to be specified there.

**Pull request closed without merging**